src/graph/entities/qg_node.c

```diff
@@ -100,6 +115,35 @@ int QGNode_GetLabelID
 	return labelId;
 }
 
+bool QGNode_IsLabelOptional
+(
+	const QGNode *n,
+	const char *label
+) {
+	ASSERT(n != NULL);
+
+	int label_count = QGNode_LabelCount(n);
+	for (uint i = 0; i < label_count; i++) {
+		if (strcmp(n->labels[i], label) == 0) {
+			return n->optional[i];
+		}
+	}
+	
+	ASSERT(false);
+	return false;
+}
+
+bool QGNode_IsLabelIdxOptional
+(
+	const QGNode *n,
+	uint idx
+) {
+	ASSERT(n != NULL);
+	ASSERT(idx < QGNode_LabelCount(n));
+
+	return n->optional[idx];
+}
+
 const char *QGNode_GetLabel
 (
 	const QGNode *n,
```

Comment:
Please update QGNode_HasLabel signature to the following:
```c
```c
bool QGNode_HasLabel
 (
 	const QGNode *n,
 	const char *label,
 	uint *idx // [OPTIONAL] set to label idx if located
 	bool *optional // [OPTIONAL]
 ) {
```
```